loader/linux/icd_linux.c

```diff
@@ -36,111 +37,168 @@ static pthread_once_t initialized = PTHREAD_ONCE_INIT;
  *
  */
 
-// go through the list of vendors in the two configuration files
-void khrIcdOsVendorsEnumerate(void)
+typedef void khrIcdFileAdd(const char *);
+
+static inline void khrIcdOsDirEntryValidateAndAdd(const char *d_name, const char *path,
+                                                  const char *extension, khrIcdFileAdd addFunc)
 {
-    DIR *dir = NULL;
-    struct dirent *dirEntry = NULL;
-    const char* vendorPath = ICD_VENDOR_PATH;
-    char* envPath = NULL;
+    struct stat statBuff;
+    char* fileName = NULL;
 
-    khrIcdVendorsEnumerateEnv();
+    // make sure the file name ends in .icd
```

Comment:
This comment became out-of-sync.
```suggestion
    // make sure the file name ends in `extension` (eg. .icd, or .lay)
```